### PR TITLE
[Data Sources] Initial commit for adding Dgraph support

### PR DESCRIPTION
**Pull request by The-Alchemist**
## What type of PR is this? (check all applicable)

- [x] Feature
- [x] New Query Runner (Data Source)

## Description
Adding support for querying [Dgraph](https://dgraph.io).

Feedback welcome!

### CODE REVIEW

File: redash/query_runner/dgraph.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        """From https://stackoverflow.com/a/2319116
+
+        Remove all occurrences streamed comments (/*COMMENT */) from string"""
+        s = re.sub(re.compile("/\*.*?\*/", re.DOTALL), "", s)
+        return s
+
+    def run_query(self, query, user):
+
+        # gotta remove comments because Dgraph uses '#' as a comment delimiter, not '/* */'
+        query = Dgraph.remove_comments(query)
```
Comment (arikfr) [inline redash/query_runner/dgraph.py]: You can remove the comment using `annotate_query`, like this:

https://github.com/getredash/redash/blob/261062d491876601f2153c8a14d1a42945d0b950/redash/query_runner/url.py#L7-L9

File: redash/query_runner/dgraph.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    import pydgraph
+    enabled = True
+except ImportError:
+    enabled = False
+
+from redash.query_runner import BaseQueryRunner, register
+from redash.query_runner import TYPE_STRING, TYPE_DATE, TYPE_DATETIME, TYPE_INTEGER, TYPE_FLOAT, TYPE_BOOLEAN
+from redash.utils import json_dumps, json_loads
+
+TYPES_MAP = {
```
Comment (arikfr) [inline redash/query_runner/dgraph.py]: This seems unused?

Comment (The-Alchemist) [reply redash/query_runner/dgraph.py]: I think I'll end up using it later for the column types, but you're right, it's unused for now so I scraped it

**Comment by The-Alchemist** [comment]
I know you're busy, @arikfr , but is there something else I can do to improve this PR?  Thanks! :)

**Comment by The-Alchemist** [comment]
Messed up rebase.  New PR is #3987

**Pull request closed without merging**